[BUG] Installing KlipperScreen inside a chroot (CustomPIOS) fails

### What happened?

During developing of a KlipperScreen Module for our Buildchain, I am not able to install KlipperScreen on Armbian based Images (they use py3.11.2). Everytime I try, it fails at the point building Ninja.

I already tried to install `ninja-build` package but this doesnt help at all.

Here is the full Output of the workflow run: https://github.com/KwadFan/MainsailOS-dev/actions/runs/7940003744/job/21680758799#step:3:21673

Building an Image using a Raspaberry Pi OS is possible and runs just fine.
As you can see here: https://github.com/mainsail-crew/MainsailOS/actions/runs/7855908040/job/21438108889#step:3:10861

So, a hint what could be worng here, would be great.
Regards Kwad
 

### What did you expect to happen instead?

Building an image ...

### How to reproduce this bug?

Fork mainsail-crew/mainsailOS, run a build workflow with enabled KlipperScreen Module.

### Additional information:

_No response_

### Log output

```shell
[16/19] Running Ninja test suite
            FAILED: build_ninja-prefix/src/build_ninja-stamp/build_ninja-run_ninja_test_suite /tmp/pip-install-t3y_ngos/ninja_611ff158dc9d4a2bb48b820cd00cfeb3/_skbuild/linux-armv7l-3.11/cmake-build/build_ninja-prefix/src/build_ninja-stamp/build_ninja-run_ninja_test_suite
            cd /tmp/pip-install-t3y_ngos/ninja_611ff158dc9d4a2bb48b820cd00cfeb3/_skbuild/linux-armv7l-3.11/cmake-build/Ninja-build && /tmp/pip-install-t3y_ngos/ninja_611ff158dc9d4a2bb48b820cd00cfeb3/_skbuild/linux-armv7l-3.11/cmake-build/Ninja-build/ninja_test && /usr/bin/cmake -E touch /tmp/pip-install-t3y_ngos/ninja_611ff158dc9d4a2bb48b820cd00cfeb3/_skbuild/linux-armv7l-3.11/cmake-build/build_ninja-prefix/src/build_ninja-stamp/build_ninja-run_ninja_test_suite
            ninja: build stopped: subcommand failed.
            Traceback (most recent call last):
              File "/tmp/pip-build-env-3spi9ese/overlay/lib/python3.11/site-packages/skbuild/setuptools_wrap.py", line 674, in setup
                cmkr.make(make_args, install_target=cmake_install_target, env=env)
              File "/tmp/pip-build-env-3spi9ese/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 697, in make
                self.make_impl(clargs=clargs, config=config, source_dir=source_dir, install_target=install_target, env=env)
              File "/tmp/pip-build-env-3spi9ese/overlay/lib/python3.11/site-packages/skbuild/cmaker.py", line 742, in make_impl
                raise SKBuildError(msg)
      
            An error occurred while building with CMake.
              Command:
                /usr/bin/cmake --build . --target install --config Release --
              Install target:
                install
              Source directory:
                /tmp/pip-install-t3y_ngos/ninja_611ff158dc9d4a2bb48b820cd00cfeb3
              Working directory:
                /tmp/pip-install-t3y_ngos/ninja_611ff158dc9d4a2bb48b820cd00cfeb3/_skbuild/linux-armv7l-3.11/cmake-build
            Please check the install target is valid and see CMake's output for more information.
      
            [end of output]
      
        note: This error originates from a subprocess, and is likely not a problem with pip.
        ERROR: Failed building wheel for ninja
      Failed to build ninja
      ERROR: Could not build wheels for ninja, which is required to install pyproject.toml-based projects
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
Unable to install dependencies, aborting install.
```

--- a/scripts/KlipperScreen-install.sh
+++ b/scripts/KlipperScreen-install.sh
@@ -108,13 +108,24 @@ create_virtualenv()
     fi
 
     source ${KSENV}/bin/activate
-    pip --disable-pip-version-check install -r ${KSPATH}/scripts/KlipperScreen-requirements.txt
+    if [[ "$(uname -m)" =~ armv[67]l ]]; then
+        echo_text "Using armv[67]l! Adding piwheels.org as extra index..."
+        pip --disable-pip-version-check install --extra-index-url https://www.piwheels.org/simple -r ${KSPATH}/scripts/KlipperScreen-requirements.txt
+    else
+        pip --disable-pip-version-check install -r ${KSPATH}/scripts/KlipperScreen-requirements.txt
+    fi
     if [ $? -gt 0 ]; then
         echo_error "Error: pip install exited with status code $?"
         echo_text "Trying again with new tools..."
         sudo apt-get install -y build-essential cmake
-        pip install --upgrade pip setuptools
-        pip install -r ${KSPATH}/scripts/KlipperScreen-requirements.txt
+        if [[ "$(uname -m)" =~ armv[67]l ]]; then
+            echo_text "Adding piwheels.org as extra index..."
+            pip install --extra-index-url https://www.piwheels.org/simple --upgrade pip setuptools
+            pip install --extra-index-url https://www.piwheels.org/simple -r ${KSPATH}/scripts/KlipperScreen-requirements.txt
+        else
+            pip install --upgrade pip setuptools
+            pip install -r ${KSPATH}/scripts/KlipperScreen-requirements.txt
+        fi
         if [ $? -gt 0 ]; then
             echo_error "Unable to install dependencies, aborting install."
             deactivate

